pkg/devfile/parser/parse_test.go

```diff
@@ -2395,3 +2398,298 @@ func Test_parseParentAndPlugin_RecursivelyReference_withMultipleURI(t *testing.T
+						},
+					},
+				},
+			},
+			wantDevFile: localDevfile,
+			uri:         localRelativeURI,
+		},
+		{
+			name:        "case 2: should be able to parse relative uri from URL",
+			curDevfile:  parentDevfile,
+			wantDevFile: relativeParentDevfile,
+			uri:         localRelativeURI,
+		},
+		{
+			name: "case 3: should fail if no path or url has been set for devfile ctx",
+			curDevfile: DevfileObj{
+				Data: &v2.DevfileV2{
+					Devfile: v1.Devfile{
+						DevWorkspaceTemplateSpec: v1.DevWorkspaceTemplateSpec{
+							Parent: &v1.Parent{
+								ImportReference: v1.ImportReference{
+									ImportReferenceUnion: v1.ImportReferenceUnion{
+										Uri: localRelativeURI,
+									},
+								},
+							},
+							DevWorkspaceTemplateSpecContent: v1.DevWorkspaceTemplateSpecContent{},
+						},
+					},
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "case 4: should fail if file not exist",
+			curDevfile: DevfileObj{
+				Ctx: devfileCtx.NewDevfileCtx(OutputDevfileYamlPath),
+				Data: &v2.DevfileV2{
+					Devfile: v1.Devfile{
+						DevWorkspaceTemplateSpec: v1.DevWorkspaceTemplateSpec{
+							Parent: &v1.Parent{
+								ImportReference: v1.ImportReference{
+									ImportReferenceUnion: v1.ImportReferenceUnion{
+										Uri: "notexist/devfile.yaml",
+									},
+								},
+							},
+							DevWorkspaceTemplateSpecContent: v1.DevWorkspaceTemplateSpecContent{},
+						},
+					},
+				},
+			},
+			uri:     "notexist/devfile.yaml",
+			wantErr: true,
+		},
+		{
+			name: "case 5: should fail if url not exist",
+			curDevfile: DevfileObj{
+				Ctx: devfileCtx.NewURLDevfileCtx(httpPrefix + uri1),
+				Data: &v2.DevfileV2{
+					Devfile: v1.Devfile{
+						DevWorkspaceTemplateSpec: v1.DevWorkspaceTemplateSpec{
+							Parent: &v1.Parent{
+								ImportReference: v1.ImportReference{
+									ImportReferenceUnion: v1.ImportReferenceUnion{
+										Uri: notExistURI,
+									},
+								},
+							},
+							DevWorkspaceTemplateSpecContent: v1.DevWorkspaceTemplateSpecContent{},
+						},
+					},
+				},
+			},
+			uri:     notExistURI,
+			wantErr: true,
+		},
+		{
+			name: "case 6: should fail if with invalid URI format",
+			curDevfile: DevfileObj{
+				Ctx: devfileCtx.NewURLDevfileCtx(OutputDevfileYamlPath),
+				Data: &v2.DevfileV2{
+					Devfile: v1.Devfile{
+						DevWorkspaceTemplateSpec: v1.DevWorkspaceTemplateSpec{
+							Parent: &v1.Parent{
+								ImportReference: v1.ImportReference{
+									ImportReferenceUnion: v1.ImportReferenceUnion{
+										Uri: invalidURL,
+									},
+								},
+							},
+							DevWorkspaceTemplateSpecContent: v1.DevWorkspaceTemplateSpecContent{},
+						},
+					},
+				},
+			},
+			uri:     invalidURL,
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// if the main devfile is from local, need to set absolute path
+			if tt.curDevfile.Ctx.GetURL() == "" {
+				err := tt.curDevfile.Ctx.SetAbsPath()
```

Comment:
how come you dont need to set it for url since we dont call Populate() for the current devfile 🤔

Reply:
because url is an absolute url already. it's being set when creating the contect `NewURLDevfileCtx`
however, when creating context with `NewDevfileCtx`, it only sets the `relPath`; so need to call an additional function `SetAbsPath()` to set the `absPath`

Reply:
we should probably update `NewDevfileCtx` next time as well so that we dont need an extra call in Populate